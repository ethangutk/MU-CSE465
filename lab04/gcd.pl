% Implement the GCD algorithm using Prolog
% See https://en.wikipedia.org/wiki/Euclidean_algorithm#Implementations

% The recursive version of he Euclidean algorithm 
% for computing the GCD  is:
%
% function gcd(a, b)
%     if b = 0
%         return a
%     else
%         return gcd(b, a mod b)


% TODO -- Define GCD here in Prolog here
gcd(A, 0, A).
gcd(A, B, R) :- AmodB is A mod B, gcd(B, AmodB, R).

% Tests
:- gcd(8, 4, X), format('gcd of 8 and 4 is ~f~n', X).
:- gcd(105, 252, X), format('gcd of 105 and 252 is ~f~n', X).
