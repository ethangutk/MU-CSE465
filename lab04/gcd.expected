gcd of 8 and 4 is 4.000000
gcd of 105 and 252 is 21.000000
